<?xml version="1.0"?>
<launch>
    <!-- 加载配置文件 -->
    <rosparam command="load" file="$(find elephant_agv_driver)/config/elephant_agv_config.yaml" />
    
    <!-- 启动AGV驱动节点 -->
    <node name="elephant_agv_node" pkg="elephant_agv_driver" type="elephant_agv_node.py" output="screen">
        <!-- 可以通过launch参数覆盖配置文件中的参数 -->
        <param name="serial_port" value="COM3" unless="$(eval arg('serial_port') == '')"/>
        <param name="baud_rate" value="115200" unless="$(eval arg('baud_rate') == '')"/>
    </node>
</launch> 